File: SPOJ/Week 1/primes/prime.c
#include<stdio.h>
#include<math.h>
int is_prime(long long n, long primes[], int j){
	double root = sqrt(n);
	long long int i = 0;
	while(primes[i] <= root && i != j){
		if( n % primes[i] == 0 )
			return 0;
		++i;
	}
	
	return 1;
}
int main(){
	long primes[9592];
	int j = 0, N;
	scanf("%d",&N);
	long long int i;
	for (i = 2; i<=100000; i++){
		if(is_prime(i, primes, j)){
			primes[j++] = i;
		}
	}
	while(N--){
		long long int m, n;
		scanf("%lld %lld",&m, &n);
		if(m<2)
			m = 2;
		i = m;
		if(i%2 == 0)
			++i;
		while(i<=n){
			if(is_prime(i, primes, 9592)){
				printf("%lld\n", i);
			}
			++i;
		}
		printf("\n");
	
	}
	return 0;
}

File: SPOJ/Week 1/primes/prime.py
from math import sqrt
from time import time
t = time()
def sieveOfErat(end):  
  if end < 2: return []  
  lng = ((end/2)-1+end%2)  
  sieve = [True]*(lng+1)  
  for i in range(int(sqrt(end)) >> 1):  
    if not sieve[i]: continue  
    for j in range( (i*(i + 3) << 1) + 3, lng, (i << 1) + 3):  
      sieve[j] = False  
  primes = [2]  
  primes.extend([(i << 1) + 3 for i in range(lng) if sieve[i]])  

  return primes  
sieveOfErat(100000)
print time() -t
'''N = input()
i = 0
while i < N:
	m,n = raw_input().split()
    m,n = int(m), int(n)
    l = 
	i+=1'''
